### PR TITLE
[main] Bump Quarkus version to 2.16.9.Final

### PR DESCRIPTION
**Pull request by kie-ci3**
Please review and merge
- https://github.com/kiegroup/kogito-pipelines/pull/1021
- https://github.com/kiegroup/kogito-images/pull/1661
- https://github.com/kiegroup/kogito-docs/pull/426
- https://github.com/kiegroup/kogito-runtimes/pull/3157
- https://github.com/kiegroup/kogito-examples/pull/1755

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#2974` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-T 1C -Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3157 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2974/display/redirect

**Test results:**
- PASSED: 3350
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2974/testReport/org.kie.kogito.tracing/QuarkusTracingAddonDevServicesIT/testEvaluateLoanEligibility/">org.kie.kogito.tracing.QuarkusTracingAddonDevServicesIT.testEvaluateLoanEligibility</a></summary>
java.lang.RuntimeException: Failed to start quarkus
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#2976` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-T 1C -Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3157 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2976/display/redirect

**Test results:**
- PASSED: 3350
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2976/testReport/org.kie.kogito.tracing/QuarkusTracingAddonDevServicesIT/testEvaluateLoanEligibility/">org.kie.kogito.tracing.QuarkusTracingAddonDevServicesIT.testEvaluateLoanEligibility</a></summary>
java.lang.RuntimeException: Failed to start quarkus
</details>

**Comment by radtriste** [comment]
jenkins rerun kogito-runtimes tests

File: kogito-build/kogito-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
   <properties>
     <!-- this version property is used in plugins but also in dependencies too -->
-    <version.io.quarkus>2.16.8.Final</version.io.quarkus>
+    <version.io.quarkus>2.16.9.Final</version.io.quarkus>
     <version.io.quarkus.quarkus-test>${version.io.quarkus}</version.io.quarkus.quarkus-test>
     <version.org.springframework.boot>2.6.6</version.org.springframework.boot>
-    <version.org.apache.kafka>3.4.0</version.org.apache.kafka>
+    <version.org.apache.kafka>3.3.2</version.org.apache.kafka>
```
Comment (radtriste) [inline kogito-build/kogito-dependencies-bom/pom.xml:22]:
```suggestion
    <version.org.apache.kafka>3.4.0</version.org.apache.kafka>
```

**Review by radtriste — APPROVED** [review]

**Pull request merged**